File: app/src/main/java/life/plank/juna/zone/data/network/model/ZoneLiveData.java
package life.plank.juna.zone.data.network.model;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.List;

import lombok.Data;

@Data
public class ZoneLiveData {
    private String liveEventType;
    private long foreignId;
    private String boardTopic;
    private String liveDataType;
    private String data;

    public LiveScoreData getScoreData(Gson gson) {
        return gson.fromJson(data, LiveScoreData.class);
    }

    public List<MatchEvent> getMatchEventList(Gson gson) {
        return gson.fromJson(data, new TypeToken<List<MatchEvent>>() {
        }.getType());
    }

    public List<Commentary> getCommentaryList(Gson gson) {
        return gson.fromJson(data, new TypeToken<List<Commentary>>() {
        }.getType());
    }

    public LiveTimeStatus getLiveTimeStatus(Gson gson) {
        return gson.fromJson(data, LiveTimeStatus.class);
    }

    public List<ScrubberData> getScrubberData(Gson gson) {
        return gson.fromJson(data, new TypeToken<List<ScrubberData>>() {
        }.getType());
    }

    public List<Highlights> getHighlightsList(Gson gson) {
        return gson.fromJson(data, new TypeToken<List<Highlights>>() {
        }.getType());
    }

    public MatchStats getMatchStats(Gson gson) {
        return gson.fromJson(data, MatchStats.class);
    }
}

File: app/src/main/java/life/plank/juna/zone/data/model/ZoneLiveData.kt
package life.plank.juna.zone.data.model

import android.os.Parcelable
import com.google.gson.Gson
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.google.gson.reflect.TypeToken
import kotlinx.android.parcel.Parcelize
import life.plank.juna.zone.data.model.notification.BaseInAppNotification

@Parcelize
data class ZoneLiveData(
        @SerializedName("foreignId") @Expose var matchId: Long,
        var boardTopic: String,
        var liveDataType: String,
        var homeTeamName: String,
        var homeTeamLogo: String,
        @SerializedName("awayTeamName") @Expose var visitingTeamName: String,
        @SerializedName("awayTeamLogo") @Expose var visitingTeamLogo: String,
        var data: String = ""
) : Parcelable, BaseInAppNotification() {

    fun getScoreData(gson: Gson): LiveScoreData? {
        return gson.fromJson<LiveScoreData>(data, LiveScoreData::class.java)
    }

    fun getMatchEventList(gson: Gson): List<MatchEvent>? {
        return gson.fromJson<List<MatchEvent>>(data, object : TypeToken<List<MatchEvent>>() {
        }.type)
    }

    fun getCommentaryList(gson: Gson): List<Commentary>? {
        return gson.fromJson<List<Commentary>>(data, object : TypeToken<List<Commentary>>() {
        }.type)
    }

    fun getLiveTimeStatus(gson: Gson): LiveTimeStatus? {
        return gson.fromJson<LiveTimeStatus>(data, LiveTimeStatus::class.java)
    }

    fun getScrubberData(gson: Gson): List<ScrubberData>? {
        return gson.fromJson<List<ScrubberData>>(data, object : TypeToken<List<ScrubberData>>() {
        }.type)
    }

    fun getHighlightsList(gson: Gson): List<Highlights>? {
        return gson.fromJson<List<Highlights>>(data, object : TypeToken<List<Highlights>>() {
        }.type)
    }

    fun getMatchStats(gson: Gson): MatchStats? {
        return gson.fromJson<MatchStats>(data, MatchStats::class.java)
    }
}